### PR TITLE
[A7] - Identity and Authentication Failures -Golang - Insecure go project - Solution

### PR DESCRIPTION
**Pull request by henriporto**
## This solution refers to which of the apps?

[A7 - Identity and Authentication Failures -Golang - Insecure go project](https://github.com/globocom/secDevLabs/tree/master/owasp-top10-2021-apps/a7/insecure-go-project)

## What did you do to mitigate the vulnerability?

Enabled authentication in mongodb using `MONGO_INITDB_ROOT_******` environment variables
([docker-entrypoint.sh checks for the existence of these variables and enable authentication.](https://github.com/docker-library/mongo/blob/master/docker-entrypoint.sh)).

Also, removed hardcoded credentials and unusable files: 
`owasp-top10-2021-apps/a7/insecure-go-project/deployments/mongo-init.js`
`owasp-top10-2021-apps/a7/insecure-go-project/deployments/mongo.Dockerfile`

## Did you test your changes? What commands did you run?

1:

**Create .env file with credentials.**

2:
```sh
Make install
```

3:
Open mongodb CLI and run:
```sh
mongo -u $MONGO_INITDB_ROOT_USERNAME -p $MONGO_INITDB_ROOT_PASSWORD --authenticationDatabase admin
```
```sh
use admin
```
```sh
show users
```

### CODE REVIEW

**Review by fguisso — COMMENTED** [review]
At this point you don't have hard code credentials anymore, this changes is just dev best practices.

File: owasp-top10-2021-apps/a7/insecure-go-project/deployments/docker-compose.yml [hunk 1]
```diff
@@ -21,15 +21,18 @@ services:
...
-        build:
-            context: ../
-            dockerfile: deployments/mongo.Dockerfile
+        image: mongo:4.0.3
+        env_file: 
+            - .env
+        environment:
+            MONGO_INITDB_ROOT_USERNAME: ${MONGO_USER}
+            MONGO_INITDB_ROOT_PASSWORD: ${MONGO_PASS}
+            MONGO_INITDB_DATABASE: ${MONGO_BASE}
```
[the remark this answers is not in the document]
Comment (henriporto) [reply owasp-top10-2021-apps/a7/insecure-go-project/deployments/docker-compose.yml]: Fixed!

File: owasp-top10-2021-apps/a7/insecure-go-project/deployments/.env.example [hunk 2]
```diff
@@ -0,0 +1,3 @@
+MONGO_INITDB_ROOT_USERNAME="u_insecure_go_project"
+MONGO_INITDB_ROOT_PASSWORD="svGX8SViufvYYNu6m3Kv"
+MONGO_INITDB_DATABASE="insecure_go_project"
```
Comment (fguisso) [inline owasp-top10-2021-apps/a7/insecure-go-project/deployments/.env.example:3]: remove the values
```
MONGO_INITDB_ROOT_USERNAME=
MONGO_INITDB_ROOT_PASSWORD=
MONGO_INITDB_DATABASE=
```

Comment (henriporto) [reply owasp-top10-2021-apps/a7/insecure-go-project/deployments/.env.example]: Done!

**Comment by fguisso** [comment]
I couldn't reproduce the attack narrative anymore! 🚀 Congratz @henriporto for securing this app and using some development best practices.

**Pull request closed without merging**